src/components/docs-version-switcher/index.tsx

```diff
@@ -57,66 +52,46 @@ const DocsVersionSwitcher = ({ options }: DocsVersionSwitcherProps) => {
 	const currentRootDocsPathName =
 		currentRootDocsPath.shortName || currentRootDocsPath.name
 	const ariaLabel = `Choose a ${currentProduct.name} ${currentRootDocsPathName} version. Currently viewing ${selectedOption.label}.`
-
 	return (
 		<nav>
-			<NavigationDisclosure
-				className={(isOpen: boolean) => {
-					return classNames(s.root, {
-						[s['root-expanded']]: isOpen,
-					})
-				}}
+			<DropdownDisclosure
+				aria-label={ariaLabel}
+				className={s.docsVersionSwitcher}
+				text={selectedOption.label}
+				color="secondary"
+				listPosition="right"
 			>
-				<NavigationDisclosureActivator
-					ariaLabel={ariaLabel}
-					className={s.activator}
-				>
-					<span className={s.activatorTextWrapper}>
-						<Text
-							asElement="span"
-							className={s.activatorText}
-							size={200}
-							weight="regular"
-						>
-							{selectedOption.label}
-						</Text>
-						<span className={s.activatorIcon}>
-							<IconCaret16 />
-						</span>
-					</span>
-				</NavigationDisclosureActivator>
-				<NavigationDisclosureContent className={s.content}>
-					<NavigationDisclosureList className={s.optionList}>
-						{options.map((option: DocsVersionSwitcherOption) => {
-							let href: string
-							if (option.isLatest) {
-								href = removeVersionFromPath(currentPath)
-							} else {
-								href = getTargetPath({
-									basePath: `${currentProduct.slug}/${currentRootDocsPath.path}`,
-									asPath: currentPath,
-									version: option.version,
-								})
-							}
-
-							return (
-								<NavigationDisclosureListItem
-									key={href}
-									className={s.optionListItem}
-								>
-									<NavigationDisclosureLink
-										className={s.optionLink}
-										href={href}
-										isActive={currentPath === href}
-									>
-										{option.label}
-									</NavigationDisclosureLink>
-								</NavigationDisclosureListItem>
-							)
-						})}
-					</NavigationDisclosureList>
-				</NavigationDisclosureContent>
-			</NavigationDisclosure>
+				<DropdownDisclosureLabelItem>
+					{currentProduct.name}
+				</DropdownDisclosureLabelItem>
+				{options
+					// Hide currently selected version from dropdown list
+					.filter(
+						(option: VersionSelectItem) =>
+							option.version !== selectedOption.version
+					)
```

Comment:
Not something actionable now, as I know this is in the design spec, but just an observation...it feels a little disorienting to have the contents of the dropdown change depending on the selected version. 💭